src/ReactiveDomain.Foundation/StreamStore/CorrelatedStreamStoreRepository.cs

```diff
@@ -35,15 +41,35 @@ public class CorrelatedStreamStoreRepository:ICorrelatedRepository
             }
         }
        
-        public TAggregate GetById<TAggregate>(Guid id, int version, ICorrelatedMessage source) where TAggregate : AggregateRoot, IEventSource {
-            var agg = _repository.GetById<TAggregate>(id, version);
+        public TAggregate GetById<TAggregate>(Guid id, int version, ICorrelatedMessage source) where TAggregate : AggregateRoot, IEventSource
+        {
+            TAggregate agg = null;
+            _cache?.GetById(id, out agg);
+
+            if(agg == null || agg.Version > version) {
+                agg = _repository.GetById<TAggregate>(id, version);
+            }
```

Comment:
It would be more explicit for this to be `if (_cache == null || agg.Version > version)`, since a null cache is only way to have `agg == null` at this point. In other words, `agg == null` at this point can only be a side effect of having `_cache == null`, so let's test for the root cause rather than the side effect.

Reply:
I'd rather not trust that the out variable could never be null, this way catch both case today and in the future.